src/java.base/share/classes/jdk/internal/access/SharedSecrets.java

```diff
@@ -335,37 +377,45 @@ public static void setJavaIORandomAccessFileAccess(JavaIORandomAccessFileAccess
     }
 
     public static JavaIORandomAccessFileAccess getJavaIORandomAccessFileAccess() {
-        if (javaIORandomAccessFileAccess == null) {
+        var access = javaIORandomAccessFileAccess;
+        if (access == null) {
             ensureClassInitialized(RandomAccessFile.class);
+            access = javaIORandomAccessFileAccess;
         }
-        return javaIORandomAccessFileAccess;
+        return access;
     }
 
     public static void setJavaSecuritySignatureAccess(JavaSecuritySignatureAccess jssa) {
         javaSecuritySignatureAccess = jssa;
     }
 
     public static JavaSecuritySignatureAccess getJavaSecuritySignatureAccess() {
-        if (javaSecuritySignatureAccess == null) {
+        var access = javaSecuritySignatureAccess;
+        if (access == null) {
             ensureClassInitialized(Signature.class);
+            access = javaSecuritySignatureAccess;
         }
-        return javaSecuritySignatureAccess;
+        return access;
     }
 
     public static void setJavaxCryptoSealedObjectAccess(JavaxCryptoSealedObjectAccess jcsoa) {
         javaxCryptoSealedObjectAccess = jcsoa;
     }
 
     public static JavaxCryptoSealedObjectAccess getJavaxCryptoSealedObjectAccess() {
-        if (javaxCryptoSealedObjectAccess == null) {
+        var access = javaxCryptoSealedObjectAccess;
+        if (access == null) {
             ensureClassInitialized(SealedObject.class);
+            access = javaxCryptoSealedObjectAccess;
         }
-        return javaxCryptoSealedObjectAccess;
+        return access;
     }
 
     private static void ensureClassInitialized(Class<?> c) {
         try {
             MethodHandles.lookup().ensureInitialized(c);
-        } catch (IllegalAccessException e) {}
+        } catch (IllegalAccessException e) {
+            throw new InternalError(e);
+        }
```

Comment:
This is a potential behavioral change that has little to do with the issue at hand, right?